tools/docker/core/Dockerfile

```
@@ -26,6 +26,10 @@ RUN apk add --no-cache --virtual .certbot-deps \
         ca-certificates \
         binutils
 
+# We set this environment variable and install git while building to try and
+# increase the stability of fetching the rust crates needed to build the
+# cryptography library
+ARG CARGO_NET_GIT_FETCH_WITH_CLI=true
```

Does `ARG` actually propagate to an environment variable?

My reading of https://docs.docker.com/engine/reference/builder/#arg is that `ARG` allows for parameterization within the scope of the Dockerfile, but you have to use `ENV` separately to make an environment variable from it.

Their example is:

```Dockerfile
ARG CONT_IMG_VER
ENV CONT_IMG_VER=${CONT_IMG_VER:-v1.0.0}
```

We're not parameterizing this for the snap, so maybe: 

```suggestion
ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
```

---

I agree Docker's documentation isn't very clear there. These two sources are a bit clearer though:

* https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg
* https://stackoverflow.com/questions/39597925/how-do-i-set-environment-variables-during-the-build-in-docker

My understanding is while you can override `ARG`, it sets an environment variable that's only available at build time which I think is what we want.

I also tested it and it seems to work:
```
cat << EOF > Dockerfile
FROM ubuntu
ARG FOO=bar
RUN env > ~/myenv
ENTRYPOINT cat ~/myenv
EOF
docker build -t foo .
docker run --rm foo
```
outputs
```
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
FOO=bar
PWD=/
```

---

TIL, thanks.